Use specialised function for decoding trust bundles

SgtCoDFish opened the pull request:
Importantly, this targets cert-manager's default (master) branch. The `beta` in cert-manager's version in go.mod comes from the `v1.16.0-beta.0` tag being the last one present on master - `v1.16.1` and `v1.16.2` appear on the release-1.16 branch and so aren't present on master.

A later release of istio-csr (presumably after we tag a beta for cert-manager 1.17) can target a tagged version of cert-manager.

SgtCoDFish commented on go.mod:
```diff
@@ -6,7 +6,7 @@ go 1.23.0
 
 require (
-	github.com/cert-manager/cert-manager v1.16.2
+	github.com/cert-manager/cert-manager v1.16.0-beta.0.0.20250107112437-c422d5944196
```
note: Confused about "beta" here? See the PR description!

wallrj reviewed (changes requested):
There are a few other places where (ithink) root CA bundles are loaded using the chain function chain function and chains are loaded using the set function:

```
$ git grep DecodeX509Certificate
...
pkg/tls/rootca/rootca.go:       rootCAsCerts, err := pki.DecodeX509CertificateChainBytes(rootCAsPEM)
...
...
test/e2e-pure-runtime/suite/suite.go:           return cmutil.DecodeX509CertificateSetBytes(chainBytes)
test/e2e/suite/runtimeconfiguration/runtimeconfiguration.go:            return cmutil.DecodeX509CertificateSetBytes(chainBytes)
```

For example, I cherry picked the regression test from #475 (a4a3dbd) and it failed:

```console
$ go test ./pkg/tls/rootca/... -run TestLoadRootCAsFile -v
=== RUN   TestLoadRootCAsFile_SystemCertPool
    rootca.go:153: I0108 14:12:01.264447] updating root CAs from file
    rootca_test.go:167:
                Error Trace:    /home/richard/projects/cert-manager/istio-csr/pkg/tls/rootca/rootca_test.go:167
                Error:          Received unexpected error:
                                failed to decode root CAs in certificate file "/etc/ssl/certs/ca-certificates.crt": provided PEM data was larger than the maximum 65000B
                Test:           TestLoadRootCAsFile_SystemCertPool
                Messages:       Should have loaded the system CA certificate pool
    rootca_test.go:168:
                Error Trace:    /home/richard/projects/cert-manager/istio-csr/pkg/tls/rootca/rootca_test.go:168
                Error:          Should be true
                Test:           TestLoadRootCAsFile_SystemCertPool
--- FAIL: TestLoadRootCAsFile_SystemCertPool (0.00s)
FAIL
FAIL    github.com/cert-manager/istio-csr/pkg/tls/rootca        0.007s
FAIL
```

SgtCoDFish commented:
@wallrj great comment, thank you! My bad for assuming there was only one place.

I've been through the code now and checked every place we import the `pki` util package. I've changed one invocation (in rootca.go) and added / tweaked some comments.
